src/NServiceBus.Core/Pipeline/PipelineModelBuilder.cs

```diff
@@ -45,8 +60,17 @@ public List<RegisterStep> Build()
                 throw new Exception(message);
             }
 
-            //  Step 2: do replacements
-            foreach (var metadata in replacements)
+            //  Step 2: validate and apply replacements
+            var groupedReplacements = replacements.GroupBy(x => x.ReplaceId);
+            if (groupedReplacements.Any(x => x.Count() > 1))
+            {
+                var duplicateReplaceIdentifiers = groupedReplacements.Where(x => x.Count() > 1).Select(x => x.Key);
+                var duplicateIdentifiersList = string.Join(",", duplicateReplaceIdentifiers);
+                var message = $"You can only replace an existing step once, '{duplicateIdentifiersList}' were detected more than once in the pipeline replacement process";
```

Comment:
I personally don't like lists in the middle of an exception message but rather see these appended to the exception message but that is personal preference.

Reply:
I like to lead with the problem detected and I also think the word process can be misleading the user, how about

>Multiple replacements of the same pipeline behaviour is not supported. Make sure that you only register a single replacement for:
> - ID1
> - ID2